### PR TITLE
port end-of-line

### PR DESCRIPTION
**Pull request by brotzeit**

### CODE REVIEW

File: rust_src/src/cmds.rs [hunk 1]
```diff
@@ -1,18 +1,70 @@
...
+    let mut newpos: isize;
+    let mut pt: isize;
+    let cur_buf = ThreadState::current_buffer();
+    loop {
+        newpos = unsafe {
+            LispObject::from(Fline_end_position(num.to_raw())).as_fixnum_or_error() as isize
+        };
+        unsafe { set_point(newpos) };
+        pt = cur_buf.pt();
+        if pt > newpos as isize && cur_buf.fetch_char(pt) == '\n' as i32 {
```
Comment (birkenfeld) [inline rust_src/src/cmds.rs]: `newpos` is already isize?

Also, `fetch_char(pt)` vs `FETCH_CHAR(pt-1)` in the original.

File: rust_src/remacs-sys/lib.rs [hunk 2]
```diff
@@ -1137,6 +1137,21 @@ extern "C" {
...
+        to: ptrdiff_t,
+        cnt: ptrdiff_t,
+        bytepos: *mut ptrdiff_t,
+    ) -> ptrdiff_t;
+    pub fn Fconstrain_to_field(
+        new_pos: Lisp_Object,
+        old_pos: Lisp_Object,
+        escape_from_edge: Lisp_Object,
+        only_in_line: Lisp_Object,
+        inhibit_caputure_property: Lisp_Object,
```
Comment (birkenfeld) [inline rust_src/remacs-sys/lib.rs]: `caputure`: typo?

Comment (shaleh) [reply rust_src/remacs-sys/lib.rs]: yup, good catch

**Comment by brotzeit** [comment]
Fixed.

File: rust_src/src/cmds.rs [hunk 3]
```diff
@@ -1,18 +1,66 @@
 //! Commands
 
 use remacs_macros::lisp_fn;
-use remacs_sys::EmacsInt;
+use remacs_sys::{EmacsInt, Fline_end_position};
```
Comment (shaleh) [inline rust_src/src/cmds.rs]: Types in one block, functions in the next. This is the pattern of the recent commits to try and reduce lines of conflict.

**Pull request merged**